### DISCUSSION TITLE
Scoped Verifications

### DISCUSSION THREAD

**Discussion by MJB222398** [VerifyTests/Verify#1119] [Ideas]
My use case for Verify is to verify the DOM of a webpage is as expected at various points in some automated tests of a web application running inside a web browser. If someone changes the DOM in some way I want to know about it and Verify is doing this well for me. However I don't want to stop further execution of the test if there is a discrepancy. I would like to be able to keep the test running and for it to report all the differences at the end. One reason for this is that a test may take 10 minutes and may verify 20 different DOMs and I don't want to have to run it 20 times accepting the changes for one DOM each time.

What I would like I think is to be able to define a 'scope' that can be used for verifications and there should be no failures thrown until the scope is disposed/stopped. 

Maybe something like:

```
using (VerificationScope.Start("DOMs"))
{
    Verify(Driver.Dom);
    .......
    Verify(Driver.Dom);
    .......
    Verify(Driver.Dom);
    .......
}
```

and

```
var scope = VerificationScope.Start("DOMs");

Verify(Driver.Dom).UnderScope(scope);
.......
Verify(Driver.Dom).UnderScope(scope);
.......
Verify(Driver.Dom).UnderScope(scope);
.......

scope.Dispose();

```

Is this something that would be useful to people?
Would it be easy to achieve?
Does it make sense as a feature?

**Comment by cmbchong** [comment]
I would also find something like this useful. My use case is using Verify.ImageHash to compare against multiple screenshots and being able to check them all at once would be a nice thing to have.

It would also be useful for getting the initial snapshots for a new test. currently when I add a new test case I have to run it multiple times to create the initial snapshots.

**Reply by SimonCropp** [reply]
happy to consider a pull request
